purldb-toolkit/tests/test_purlcli.py

```diff
@@ -101,10 +103,22 @@ def test_metadata_cli(self):
             assert output == expected
 
         # NOTE: To avoid errors from the addition of new versions, we exclude
-        # `(output_data["packages"], expected_data["packages"])` from the
-        # result_objects list above and handle here.
-        for expected in expected_data["packages"]:
-            assert expected in output_data["packages"]
+        # "packages" from the result_objects list above and handle here.
+        # All other live-fetch tests are handled in a similar manner.
+
+        expected_metadata = []
```

Comment:
I see that you have used the code from line 109 to 121 in different places in this file. I think it would be better if put this in a function and then call it in those places.

Reply:
Thanks for this @JonoYang -- my new function `compare_packages(expected_data, actual_data, nested_key)` replaces code blocks for 3 `metadata` tests and 3 `versions` tests.